tests/phpunit/test-wc-stripe-level-3-data.php

```diff
@@ -125,4 +225,38 @@ public function test_non_us_shipping_zip_codes() {
 			$result
 		);
 	}
+
+	public function test_full_level3_data_with_fee() {
+		$expected_data = array(
+			'merchant_reference'   => '210',
+			'shipping_amount'      => 3800,
+			'line_items'           => array(
+				(object) array(
+					'product_code'        => 30,
+					'product_description' => 'Beanie with Logo',
+					'unit_cost'           => 1800,
+					'quantity'            => 1,
+					'tax_amount'          => 270,
+					'discount_amount'     => 0,
+				),
+				(object) array(
+					'product_code'        => 'fee',
+					'product_description' => 'fee',
+					'unit_cost'           => 1000,
+					'quantity'            => 1,
+					'tax_amount'          => 150,
+					'discount_amount'     => 0,
+				),
+			),
+			'shipping_address_zip' => '98012',
+			'shipping_from_zip'    => '94110',
+		);
+
+		update_option( 'woocommerce_store_postcode', '94110' );
+
+		$mock_order   = $this->mock_level_3_order( '98012', true );
+		$level_3_data = $this->wcpay_gateway->get_level3_data_from_order( $mock_order );
```

Comment:
I think you meant to call the Stripe gateway here using something like

```
$gateway = new WC_Gateway_Stripe();
$gateway->get_level3_data_from_order( $mock_order );
```

Reply:
Right, I accidentally committed a draft test suite (copied it from the WCPay). I am going to adjust it as soon as I can run tests locally.

Reply:
Fixed